x/sequencer/types/metadata.go

```diff
@@ -38,30 +38,78 @@ func (d SequencerMetadata) UpdateSequencerMetadata(d2 SequencerMetadata) (Sequen
 	return metadata.EnsureLength()
 }
 
-func (d SequencerMetadata) Validate() error {
+func (d SequencerMetadata) Validate(isEVM bool) error {
 	_, err := d.EnsureLength()
 	if err != nil {
 		return err
 	}
 
-	if err = d.ContactDetails.Validate(); err != nil {
+	if err = d.validateRPCs(); err != nil {
 		return err
 	}
 
-	return nil
+	if isEVM {
+		if err = d.ValidateEVMRPCs(); err != nil {
+			return err
+		}
+	}
+
+	if d.ContactDetails == nil {
+		return nil
+	}
+
+	return d.ContactDetails.Validate()
 }
 
 func (cd ContactDetails) Validate() error {
 	if err := validateURL(cd.Website); err != nil {
-		return errorsmod.Wrap(ErrInvalidURL, err.Error())
+		return errorsmod.Wrap(ErrInvalidURL, "invalid website URL")
```

Comment:
why u don't use the returned `err`?

Reply:
I found it more informative to know which field was invalid, than why